### ISSUE TITLE
Query: order by anonymous object is silently ignored

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFrameworkCore#11525]
query:

```
ctx.Squads.OrderByDescending(s => new { s.Name })
```

produces the following QM:

```
  (QueryContext queryContext) => IEnumerable<Squad> _FastQuery(
      relationalQueryContext: (RelationalQueryContext)queryContext, 
      shaperCommandContext: SelectExpression: 
          SELECT [s].[Id], [s].[InternalNumber], [s].[Name]
          FROM [Squads] AS [s]
          ORDER BY (SELECT 1) DESC, 
      materializer: (DbDataReader dataReader) => 
      {
          instance = new Squad()
          instance.<Id>k__BackingField = bool dataReader.IsDBNull(0) ? default(int) : try { int dataReader.GetInt32(0) } catch (Exception) { ... } 
          instance.<InternalNumber>k__BackingField = bool dataReader.IsDBNull(1) ? default(int) : try { int dataReader.GetInt32(1) } catch (Exception) { ... } 
          instance.<Name>k__BackingField = bool dataReader.IsDBNull(2) ? default(string) : try { string dataReader.GetString(2) } catch (Exception) { ... } 
          return instance
      }
      , 
      contextType: TestModels.GearsOfWarModel.GearsOfWarContext, 
      logger: DiagnosticsLogger<Query>)
```

**Comment by divega** [comment]
@maumar to fix this by generating OrderBy().ThenBy() for GroupBy with Include.

**Comment by smitpatel** [comment]
We decided to do client eval on OrderBy Anonymous object. It will work as per linq specs.

**Comment by maumar** [comment]
fixed in 3902d943a489107d9e2336a434ecf489376d0516
